packages/desktop/src/services/FileDownloadServiceElectron.ts

```diff
@@ -448,15 +460,27 @@ export default class FileDownloadServiceElectron extends FileDownloadService {
                         onProgress(fileDownloadedBytes);
                     }
                 });
+
+                // If cancel was requested, cleanup.
+                if (cancelRequested) {
```

Comment:
This checks if cancel was requested during the work from line 448 to 462?  Otherwise we'd just check again at the start of the next item in the 'for'?